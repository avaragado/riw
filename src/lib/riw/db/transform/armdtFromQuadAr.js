// @flow

import map from 'ramda/src/map';

import { sDescriptionDefault } from '../../';

type Transformer = (quads: RIWDBQuad[]) => RIWTranslatedMessageDescriptor[];

const fromQuad = (quad: RIWDBQuad): RIWTranslatedMessageDescriptor => ({
    defaultMessage: quad[0],
    locale: quad[2],
    translation: quad[3],
    ...(quad[1] === sDescriptionDefault ? null : { description: quad[1] }),
});

const fromQuads: Transformer = map(fromQuad);

export default fromQuads;
