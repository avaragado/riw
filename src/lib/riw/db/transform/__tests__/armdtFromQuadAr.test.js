// @flow

import type { TranslationQuad } from '../../../../../types';
import armdtFromQuadAr from '../armdtFromQuadAr';

type Fixture = {
    name: string,
    in: TranslationQuad[],
};

const fixtures: Fixture[] = [
    {
        name: '01 no quads',
        in: [],
    },

    {
        name: '02 some quads',
        in: [
            ['one', 'desc1', 'aa-aa', '[aa-aa]1 one'],
            ['one', 'desc2', 'aa-aa', '[aa-aa]2 one'],
            ['one', 'desc1', 'bb-bb', '[bb-bb]1 one'],
            ['one', 'desc2', 'bb-bb', '[bb-bb]2 one'],
            ['two', 'desc1', 'aa-aa', '[aa-aa]1 two'],
            ['two', 'desc1', 'bb-bb', '[bb-bb]1 two'],
            ['three', 'desc1', 'aa-aa', '[aa-aa]1 three'],
            ['three', 'desc1', 'cc-cc', '[cc-cc]1 three'],
            ['four', '_', 'aa-aa', '[aa-aa]no desc four'],
            ['four', 'desc1', 'aa-aa', '[aa-aa]1 four'],
        ],
    },
];

describe('lib/riw/db/transform/armdtFromQuadAr', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            const received = armdtFromQuadAr(fixture.in);

            expect(received).toMatchSnapshot();
        });
    });
});
