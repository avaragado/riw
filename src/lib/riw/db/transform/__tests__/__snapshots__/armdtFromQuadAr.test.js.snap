// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/riw/db/transform/armdtFromQuadAr 01 no quads 1`] = `Array []`;

exports[`lib/riw/db/transform/armdtFromQuadAr 02 some quads 1`] = `
Array [
  Object {
    "defaultMessage": "one",
    "description": "desc1",
    "locale": "aa-aa",
    "translation": "[aa-aa]1 one",
  },
  Object {
    "defaultMessage": "one",
    "description": "desc2",
    "locale": "aa-aa",
    "translation": "[aa-aa]2 one",
  },
  Object {
    "defaultMessage": "one",
    "description": "desc1",
    "locale": "bb-bb",
    "translation": "[bb-bb]1 one",
  },
  Object {
    "defaultMessage": "one",
    "description": "desc2",
    "locale": "bb-bb",
    "translation": "[bb-bb]2 one",
  },
  Object {
    "defaultMessage": "two",
    "description": "desc1",
    "locale": "aa-aa",
    "translation": "[aa-aa]1 two",
  },
  Object {
    "defaultMessage": "two",
    "description": "desc1",
    "locale": "bb-bb",
    "translation": "[bb-bb]1 two",
  },
  Object {
    "defaultMessage": "three",
    "description": "desc1",
    "locale": "aa-aa",
    "translation": "[aa-aa]1 three",
  },
  Object {
    "defaultMessage": "three",
    "description": "desc1",
    "locale": "cc-cc",
    "translation": "[cc-cc]1 three",
  },
  Object {
    "defaultMessage": "four",
    "locale": "aa-aa",
    "translation": "[aa-aa]no desc four",
  },
  Object {
    "defaultMessage": "four",
    "description": "desc1",
    "locale": "aa-aa",
    "translation": "[aa-aa]1 four",
  },
]
`;
